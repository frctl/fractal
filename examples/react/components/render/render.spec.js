const { render } = require('@frctl/adapter-tests');

const fractal = require('../../fractal.config.js');

describe('render', () => {
    beforeEach(async () => {
        await fractal.load();
    });

    render(fractal);

    it('renders empty string if ssr is turned off', async () => {
        fractal.components.engine(require('@frctl/react')({ ssr: false }));
        const render = await fractal.components.find('@render').render();
        expect(render).toEqual('');
    });

    it('renders component if ssr is turned off but enabled in component meta', async () => {
        fractal.components.engine(require('@frctl/react')({ ssr: false }));
        const render = await fractal.components.find('@render').render(undefined, { ssr: true });
        expect(render).toMatchSnapshot();
    });

    it('renders component with renderToStaticMarkup method', async () => {
        fractal.components.engine(require('@frctl/react')({ renderMethod: 'renderToStaticMarkuo' }));
        const render = await fractal.components.find('@render').render();
        expect(render).toMatchSnapshot();
    });

    it('renders component using ES6 import/export syntax', async () => {
        const render = await fractal.components.find('@render--es6-import-export').render();
        expect(render).toMatchSnapshot();
    });
});
