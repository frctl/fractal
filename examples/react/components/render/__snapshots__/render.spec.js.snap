// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders component if ssr is turned off but enabled in component meta 1`] = `"Render <!-- -->something"`;

exports[`render renders component with renderToStaticMarkup method 1`] = `"Render something"`;

exports[`render renders default template for default variant 1`] = `"Render <!-- -->something"`;

exports[`render renders default template for regular variant 1`] = `"Render <!-- -->variant-1"`;

exports[`render renders specified camelCased template for variant 1`] = `"Render camelCaseVariant <!-- -->something"`;

exports[`render renders specified template for variant 1`] = `"Render variant-2 <!-- -->something"`;

exports[`render renders variant template for variant with the same name 1`] = `"Render variant-2 <!-- -->something"`;
